src/DesignMyNight/Elasticsearch/Connection.php

```diff
@@ -376,6 +378,41 @@ public function pretend(Closure $callback)
 
     }
 
+    /**
+     * Get the timeout for the entire Elasticsearch request
+     * @return float
+     */
+    public function getRequestTimeout(): float
+    {
+        return $this->requestTimeout;
+    }
+
+    /**
+     * Get the timeout for the entire Elasticsearch request
+     * @param float $requestTimeout
+     * @return self
+     */
+    public function setRequestTimeout(float $requestTimeout): self
+    {
+        $this->requestTimeout = $requestTimeout;
+
+        return $this;
+    }
+
+    /**
+     * Add client-specific parameters to the request params
+     * @param array $params
+     * @return array
+     */
+    protected function addClientParams(array $params): array
+    {
+        if ($this->requestTimeout){
+            $params['client']['timeout'] = $this->requestTimeout;
```

Comment:
I imagine client is always set so there's no need to check here? e.g. isset($params['client'])

Reply:
JS would moan but PHP just adds it if it's empty so I think we're fine here.